strutil/strutil.go

```diff
@@ -40,3 +42,38 @@ func JSON(v interface{}) string {
 	bytes, _ := json.Marshal(v)
 	return string(bytes)
 }
+
+func wildCardToRegexp(pattern string) string {
```

Comment:
Could you write unittest for this function?

Reply:
What are your thoughts on using `url.Url.Match()` instead of implementing another wildcard to regex logic? Depending on source, we could create a new remote or local `url.URL` using source URL's bucket, prefix and given exclude patterns, then use its `Match` method.

Local to remote with a directory like this;
```
testdir/
├── file.txt
├── file1.txt
├── main
│   └── file.txt
├── main.js
├── main.pdf
├── main.py
└── readme.md
```

```
s5cmd cp --exclude 'main*' testdir s3://bucket/prefix/
exclude urls: [testdir/main*]

url: testdir/file.txt src, exclude url: testdir/main*, should-exclude: false
url: testdir/file1.txt src, exclude url: testdir/main*, should-exclude: false
url: testdir/main/file.txt src, exclude url: testdir/main*, should-exclude: true
url: testdir/main.js src, exclude url: testdir/main*, should-exclude: true
url: testdir/main.pdf src, exclude url: testdir/main*, should-exclude: true
url: testdir/main.py src, exclude url: testdir/main*, should-exclude: true
url: testdir/readme.md src, exclude url: testdir/main*, should-exclude: false
```

Same files on remote to local;
```
s3://bucket/
	prefix/testdir/file.txt
	prefix/testdir/file1.txt
	prefix/testdir/main.js
	prefix/testdir/main.pdf
	prefix/testdir/main.py
	prefix/testdir/main/file.txt
	prefix/testdir/readme.md
```

```
s5cmd cp --exclude 'main*' 's3://bucket/prefix/testdir/*' testdir
exclude urls: [s3://bucket/testdir/testdir/main*]

url: s3://bucket/prefix/testdir/file.txt src, exclude url: s3://bucket/prefix/testdir/main*, should-exclude: false
url: s3://bucket/prefix/testdir/file1.txt src, exclude url: s3://bucket/prefix/testdir/main*, should-exclude: false
url: s3://bucket/prefix/testdir/main.js src, exclude url: s3://bucket/prefix/testdir/main*, should-exclude: true
url: s3://bucket/prefix/testdir/main.pdf src, exclude url: s3://bucket/prefix/testdir/main*, should-exclude: true
url: s3://bucket/prefix/testdir/main.py src, exclude url: s3://bucket/prefix/testdir/main*, should-exclude: true
url: s3://bucket/prefix/testdir/main/file.txt src, exclude url: s3://bucket/prefix/testdir/main*, should-exclude: true
url: s3://bucket/prefix/testdir/readme.md src, exclude url: s3://bucket/prefix/testdir/main*, should-exclude: false
```

Reply:
Wrote unit tests for the wildcardtoRegex function. It is in `strutil_test.go`.  Will try to work `--exclude` with URL as @seruman suggested.